### DISCUSSION TITLE
How does wagmi get a short error message?

### DISCUSSION THREAD

**Discussion by the-content** [wevm/wagmi#2865] [General]
How to get the custom thrown error in the contract when reading and writing, for example, in solidity, there is a require(condition, "Insufficient distributable balance"),I want to get the error message in onError in wagmi, but when I get the error A lot of them like the following,I need to get the error message ` User rejected the request.`,how should I get the short error message?
I tried `error.message`, and I still get a lot of messages like this
```bash
writeError: TransactionExecutionError: User rejected the request.

Request Arguments.
  from: 0x000...0001
  to: 0x000...0002
  value: 0 ETH
  value: 0 ETH   0x095ea7b300000000000000000000000000007392a17feac9a74938834bb486be7e6de6686810ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff

Details: MetaMask Tx Signature: User denied transaction signature.
Version: viem@1.4.2
    at getTransactionError (getTransactionError.js:7:1)
    at sendTransaction (sendTransaction.js:121:34)
    at async writeContract (writeContract.js:59:1)
    at async writeContract (chunk-LAFZBYO7.js:2357:1)
``

**Comment by jxom** [comment]
Try `error.shortMessage`!

**Reply by the-content** [reply]
Thanks a lot, solved my problem!

**Reply by the-content** [reply]
I have this problem, I can get the error ```User rejected the request.```, when I cancel the transaction using shortMessage, but when I execute the contract write method, I throw the revert("Amount is lower than the setting"), in the contract. But ```shortMessage``` gets ```An internal error was received.```,it doesn't catch the specific error message in the revert, here is the error message from the console:
```
ContractFunctionExecutionError: An internal error was received.

URL: http://127.0.0.1:8545
Request body: {"method": "eth_call", "params":[{"from": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266", "data":"  value": "0x0"}, "latest"]}
 
Raw Call Arguments.
  from: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
  to: 0xAEb4c1F32F671F8cb671164f8e50867e3DAd180b
  value: 0 ETH
  
 
Contract Call.
  address: 0xAEb4c1F32F671F8cb671164f8e50867e3DAd180b
  function: withdraw(uint256 _withdrawType)
  function: withdraw(uint256 _withdrawType)
  sender: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266

Docs: https://viem.sh/docs/contract/simulateContract.html
Details: Error: VM Exception while processing transaction: reverted with reason string 'Amount is lower than the setting'
Version: viem@1.4.2
```

**Reply by jxom** [reply]
Are you using Anvil, Hardhat or Ganache?

**Reply by the-content** [reply]
Yes, I'm using local hardhat and fork BSC Mainnet

**Reply by the-content** [reply]
Is it because I'm using hardhat?

**Reply by wulaizi** [reply]
> 是因为我戴了安全帽吗？

Hello, have you solved it? I encountered the same problem and wanted to ask how you resolved it.

**Comment by yashsisodiya** [comment]
Did you find any solution to get short messages

**Reply by jxom** [reply]
https://github.com/wevm/wagmi/discussions/2865#discussioncomment-6666163

**Reply by yashsisodiya** [reply]
Thanks
